courses/fundamentals_of_ada/060_record_types.rst

```diff
@@ -621,121 +547,119 @@ Quiz
 
-* *Variant record type* is a record type where
+ A. :ada:`C => 100`
+ B. Multiple declaration calls :ada:`Next` twice
+ C. Correct
+ D. :ada:`C => 100` ha no effect on :ada:`A` and :ada:`B`
 
-   + Different objects may have different sets of components (i.e. different variants)
-   + Given object itself may be *unconstrained*
+=======================
+Discriminated Records
+=======================
 
-      * Different variants at different times
+----------------------------
+Discriminated Record Types
+----------------------------
 
-* Supported in other languages
+* **Discriminated** record type
 
-   + Variant records in Pascal
-   + Unions in C
+   + Different **objects** may have **different** components
+   + All object **still** share the same type
 
-* Variant record offers a kind of storage overlaying
+* Kind of **storage overlay**
 
-   + Same storage might be used for one variant at one time, and then for another variant later
-   + Language issue: Ensure this does not provide loophole from type checking
+   + Similar to :C:`union` in C
+   + But preserves **type checking**
+   + And object size **depends** on discriminant
 
-      * Neither Pascal nor C avoids this loophole
+* Aggregate assignment is allowed
 
--------------------------------------
-Discriminant in Ada Variant Records
--------------------------------------
+---------------
+Discriminants
+---------------
 
 .. code:: Ada
 
-  type Person_Tag is (Student, Faculty);
-  type Person (Tag : Person_Tag) is record
+  type Person_Group is (Student, Faculty);
+  type Person (Group : Person_Group) is record
      Name : String (1 .. 10);
-     case Tag is
+     case Group is
         when Student => -- 1st variant
            Gpa  : Float range 0.0 .. 4.0;
-           Year : Integer range 1 .. 4;
         when Faculty => -- 2nd variant
            Pubs : Integer;
      end case;
   end record;
 
-* Variant record type contains a special field (*discriminant*) whose value indicates which variant is present
-* When a field in a variant is selected, run-time check ensures that discriminant value is consistent with the selection
+* :ada:`Group` is the **discriminant**
+* Run-time check for component **consistency**
 
-   + If you could store into `Pubs` but read `GPA`, type safety would not be guaranteed
+   + eg :ada:`A_Person.Pubs := 1` checks :ada:`A_Person.Group = Faculty`
+   + :ada:`Constraint_Error` if check fails
 
-* Ada prevents this type of access
+* Discriminant is **constant**
 
-   + Discriminant (Tag) established when object of type Person created
-   + Run-time check verifies that field selected from variant is consistent with discriminant value
+   + Unless object is **mutable**
 
-      * Constraint_Error raised if the check fails
+------------------------------
+Mutable Discriminated Record
+------------------------------
+
+* When discriminant has a **default value**
+
+   + Any object instanciated **without** a discriminant value is **mutable**
```

Comment:
instantiated

Reply:
Not true - if the discriminant has a default, the object is mutable (if you create a derived type and specify the discriminant, then objects of _that_ type are immutable)